app/src/main/java/com/example/atheneum/activities/BookInfoActivity.java

```diff
@@ -0,0 +1,143 @@
+    private TextView textDesc;
+    private TextView textStatus;
+
+    private Button deleteBtn;
+    private Button editBtn;
+
+    private static final String TAG = "viewBookInfo";
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_book_info);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+
+        bookID = getIntent().getStringExtra("bookID");
+        Log.i("bookid value: ", bookID);
+        textTitle = (TextView) findViewById(R.id.bookTitle);
+        textAuthor = (TextView) findViewById(R.id.bookAuthor);
+        textIsbn = (TextView) findViewById(R.id.bookISBN);
+        textDesc = (TextView) findViewById(R.id.bookDescription);
+        textStatus = (TextView) findViewById(R.id.bookStatus);
+
+        deleteBtn = findViewById(R.id.buttonDelete);
+        deleteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                deleteBook();
+            }
+        });
+        editBtn = findViewById(R.id.buttonEdit);
+        //TODO: implement edit book button
+
+        retrieveBookInfo();
+
+    }
+
+    public void retrieveBookInfo(){
+        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
+        final FirebaseDatabase db = FirebaseDatabase.getInstance();
+        DatabaseReference ref = db.getReference().child("ownerCollection").child(firebaseUser.getUid()).
+                child(bookID);
+        if(firebaseUser != null){
+            ref.addValueEventListener(new ValueEventListener() {
+                @Override
+                public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                    Book book = dataSnapshot.getValue(Book.class);
+                    textTitle.setText(book.getTitle());
+                    textAuthor.setText(book.getAuthor());
+                    textIsbn.setText(String.valueOf(book.getIsbn()));
+                    textDesc.setText(book.getDescription());
+                    textStatus.setText(String.valueOf(book.getStatus()));
+                }
+
+                @Override
+                public void onCancelled(@NonNull DatabaseError databaseError) {
+                    System.out.println("The read failed: " + databaseError.getCode());
+                }
+            });
+
+        }
+
+    }
+
+
+    public void deleteBook(){
+        Log.i(TAG, "Delete book button pressed");
+
+        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
+        final FirebaseDatabase db = FirebaseDatabase.getInstance();
+        DatabaseReference ref = db.getReference().child("ownerCollection"). child(firebaseUser.getUid());
```

Comment:
Possible NullPointerException. Call firebaserUser.getUid after the firebaseUser != null check. This must be fixed before merged!

Reply:
You could end up doing dangerous things since this query would be wrong if the user is null.